src/providers/ProviderInterface.ts

```diff
@@ -0,0 +1,5 @@
+import type { ConstructInterface } from "@lift/constructs";
+
+export interface ProviderInterface {
+    create(type: string, id: string): ConstructInterface;
```

Comment:
This seems to conflict with the static `create()` method. Maybe we should name it `createConstruct()`?